### ISSUE TITLE
Can't turn off tooltips in window list

### ISSUE THREAD

**Issue by jhalpin100** [mate-desktop/mate-panel#534]
I keep getting tooltips in the window list widget, and can't turn them
  off. I've used dconf-editor to turn off every entry it has for
  tooltips, but I still see them for the window list.

  When I say "window list" I'm referring to the item in the "Add to
  Panel" dialog.

  In dconf-editor, the org->mate->panel->general entry shows tooltips-
  enabled = false. Is there a problem with dconf-editor? Maybe it's not
  communicating the setting to the underlying system or something?

  Thanks

  Joe

**Comment by muesli4** [comment]
> Is there a problem with dconf-editor?

No. The gsettings key seems to be used only by very few panel applets (e.g. menu). The window list is based on libwnck's tasklist which doesn't support disabling tooltips.

**Comment by minuxlintebiandedition** [comment]
May I please ask for this issue to be reopened. It is still true that there is no obvious way to disable these tooltips but it can be done.
Please look at Linux Mint Debian Edition 3 or 4, these use mate panel with cinnamon, the window list, and nemo-preview.
If nemo-preview is used with mate-desktop, the tooltips clash with the nemo-preview.

The tooltips do not pay any heed to the desktop theme, and are starting to look dated. May I suggest having a 'disable windowlist tooltips' tickbox option in the mate-panel configuration dialog?

**Comment by mbkma** [comment]
This issue still is open.
I do not know cinnamon well, but I think they use their own panel: https://github.com/linuxmint/cinnamon/blob/master/js/ui/panel.js
I remember looking in the code of libwnck to make a disable tooltip setting, but I found it very complicated to get it working reliably.

**Comment by raveit65** [comment]
This is part of libwnck.
A while ago i use this patch for personal use which disables tooltips.
```
From d7dea81f5b8418aa9b8e2d44a1edbe7509706f10 Mon Sep 17 00:00:00 2001
From: raveit65 <mate@raveit.de>
Date: Sun, 14 Jun 2020 18:47:41 +0200
Subject: [PATCH] task-list: disable tooltips

---
 libwnck/tasklist.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libwnck/tasklist.c b/libwnck/tasklist.c
index 2b20424..85ff463 100644
--- a/libwnck/tasklist.c
+++ b/libwnck/tasklist.c
@@ -3898,6 +3898,8 @@ wnck_task_create_widgets (WnckTask *task, GtkReliefStyle relief)
   gtk_widget_set_tooltip_text (task->button, text);
   g_free (text);
 
+  gtk_widget_set_has_tooltip (task->button, FALSE);
+
   /* Set up signals */
   if (GTK_IS_TOGGLE_BUTTON (task->button))
     g_signal_connect_object (G_OBJECT (task->button), "toggled",
-- 
2.26.2
```
So this is not directly in our hand. 
In theory libwnck needs a function to enable/disable tooltips which we can call from mate-panel code.
But i can't do that.

**Comment by minuxlintebiandedition** [comment]
I would love to use this 'patch' but I have no idea where to put it or what to do with it.

**Comment by minuxlintebiandedition** [comment]
![Screenshot from 2022-03-06 10-43-17](https://user-images.githubusercontent.com/42706563/156933337-dd4579a5-5833-415a-bf99-b72d21e33752.png)

**Comment by muesli4** [comment]
> I would love to use this 'patch' but I have no idea where to put it or what to do with it.

As said countless times here, it is part of **libwnck**.

**Comment by minuxlintebiandedition** [comment]
I assume you mean adding that text to the source code and compiling it?

Or putting it in some config file?

**Comment by lukefromdc** [comment]
That is c code to be patched into a source code file and built

**Comment by minuxlintebiandedition** [comment]
> That is c code to be patched into a source code file and built

Ok

**Comment by linuxCowboy** [comment]
This long-standing issue has been annoying me, too.

The culprit is the **libwnck** library.

_Window Navigator Construction Kit_

Get the source in your distro
```
# debian
apt-get source libwnck3
```
patch _tasklist.c_

`sed -i '/gtk_widget_set_tooltip_text/ s/ text/ NULL/' libwnck/tasklist.c`

compile, copy to _libdir_ and replace the link

e.g.
`sudo ln -sf libwnck-3.so.0.3.0.fix libwnck-3.so.0`

and the tooltips are __gone for good__!

You can test with

`./test-tasklist`

and find the exact library name with

`ldd /usr/lib/mate-panel/wnck-applet |grep wnck`
or
`ldd test-tasklist |grep wnck`

**Comment by muktupavels** [comment]
https://gitlab.gnome.org/GNOME/libwnck/-/merge_requests/49

**Comment by raveit65** [comment]
Hmm, i just tested 
https://gitlab.gnome.org/GNOME/libwnck/-/commit/6aa863b5825dd9dfbdbb0e60f52c598f1ba959e8
and
https://gitlab.gnome.org/GNOME/libwnck/-/commit/c55f49e18710e5d26cf5cc7301b140d1dc73b7c0
But it doesn't work very well. With most windows the tooltip are there. eg.
![tooltips](https://user-images.githubusercontent.com/961604/195984654-c2578449-4ea5-4177-ad85-fab9ae2001cc.jpg)
Is there a work with mate-panel needed to see the option in panel settings?

**Comment by muktupavels** [comment]
Automatically are hidden only useless tooltips. To completely disable tooltips you have to use API - `wnck_tasklist_set_tooltips_enabled`.

**Comment by raveit65** [comment]
OK, got it
